packages/core-mobile/app/services/walletconnectv2/WalletConnectService.ts

```diff
@@ -73,8 +74,18 @@ class WalletConnectService implements WalletConnectServiceInterface {
 
     this.client.on('session_request', requestEvent => {
       const requestSession = this.getSession(requestEvent.topic)
-      requestSession &&
+      if (requestSession) {
+        /**
+         * Solana dApps use different parameter formats than our internal
+         * VM module so we need to transform them, @see solanaRequestUtils.ts
+         * for more details
+         */
+        if (requestEvent.params.request.method.includes('solana_')) {
+          transformSolanaParams(requestEvent, requestSession)
+        }
```

Comment:
I don't think we should do this transform here since wallet connect service doesn't/shouldn't know about the vm module stuff. it would be great if this is done in the rpc listeners instead before the request is passed to vm module.